hollow/src/main/java/com/netflix/hollow/api/consumer/metrics/AbstractRefreshMetricsListener.java

```diff
@@ -55,11 +56,15 @@ public abstract class AbstractRefreshMetricsListener extends AbstractRefreshList
     ConsumerRefreshMetrics.Builder refreshMetricsBuilder;
 
     private final Map<Long, Long> cycleVersionStartTimes;
+    private final Map<Long, Long> announcementTimestamps;
```

Comment:
There is possibility of concurrent modification to this map from the refresh thread and the thread that calls `fireAnnouncementDetected` because of the way we've wired up the announcement metadata/ispinned propagtation. Lets peer on fixing this- and try to maintain the norm of all listeners invoked from only the refresh thread.

Reply:
Thanks for pointing this out. As discussed in our peering sessions, this concern is now addressed by change in HollowClientUpdater's logic. Now the announcement timestamps are recorded using the same thread which also triggers refresh started and completion.